compiler/grainc/dune

```diff
@@ -3,4 +3,4 @@
  (public_name grainc)
  (package grain)
  (modules grainc)
- (libraries grain binaryen.native dune-build-info))
+ (libraries grain lsp binaryen.native dune-build-info))
```

Comment:
Since we are splitting this out now, I propose we rename it to `diagnostics` because I foresee us building a lot of the tooling into this.

Reply:
Agreed, I'll do that.

Reply:
Renamed to Grain_diagnosics to avoid any potential conflicts.